### PR TITLE
DT-191 Prevent creation of case notes for types that are not active.

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/casenotes/controllers/ControllerAdvice.java [hunk 1]
```diff
@@ -51,6 +52,17 @@
                         .build());
     }
 
+    @ExceptionHandler(ValidationException.class)
+    public ResponseEntity<ErrorResponse> handleException(final ValidationException e) {
+        log.debug("Bad Request (400) returned", e);
+        return ResponseEntity
+                .status(HttpStatus.BAD_REQUEST)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/casenotes/controllers/ControllerAdvice.java]: there is a `badRequest()` method

File: src/main/java/uk/gov/justice/hmpps/casenotes/controllers/ControllerAdvice.java [hunk 2]
```diff
@@ -51,6 +52,17 @@
...
+                .body(ErrorResponse
+                        .builder()
+                        .status(HttpStatus.BAD_REQUEST.value())
+                        .build());
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/hmpps/casenotes/controllers/ControllerAdvice.java]: not including any error messsage in the response?

Comment (petergphillips) [reply src/main/java/uk/gov/justice/hmpps/casenotes/controllers/ControllerAdvice.java]: although will be confusing as we already catch `Exception` and return `BAD_REQUEST` with a developer message so would be worth putting one in...

Comment (andrewrlee) [reply src/main/java/uk/gov/justice/hmpps/casenotes/controllers/ControllerAdvice.java]: Catching exception and returning bad request looks a bit wrong - shouldn't a generic error return internal server error?

Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/casenotes/controllers/ControllerAdvice.java]: Agreed!

**Pull request merged**